### PR TITLE
Handling requests for content within routes

### PR DESCRIPTION
**Pull request by kevindew**
This changes the GET content and GET publish-intent endpoints so that they can accept a request for content which is within the routes of an item. For both cases if the path matches a route and not a base_path a 303 redirect will be sent to the route with a base_path.

When this is deployed we need to update indexes for mongoid.

Unfortunately a bunch of dependencies were needed to be updated to get this all working and they're somewhat tangled in this.

### CODE REVIEW

File: app/models/publish_intent.rb [hunk 1]
```diff
@@ -58,6 +58,12 @@ def self.cleanup_expired
     where(:publish_time.lt => PUBLISH_TIME_LEEWAY.ago).delete_all
   end
 
+  def base_path_without_root
```
Comment (thomasleese) [inline app/models/publish_intent.rb]: This could be changed to `base_path.sub(%r{^/}, "") if base_path`.

File: lib/tasks/pact.rake [hunk 2]
```diff
@@ -17,7 +17,7 @@ unless Rails.env.production?
     end
   end
 
-  task default: "pact:verify"
+  # task default: "pact:verify"
```
Comment (thomasleese) [inline lib/tasks/pact.rake]: Will we add a card or put a FIXME comment to avoid forgetting about this?

Comment (kevindew) [inline lib/tasks/pact.rake]: good point, I've included it in acceptance criteria of https://trello.com/c/qjoezykg/901-reinstate-the-pact-broker-2

File: app/models/publish_intent.rb (hunk 1 above)
Comment (kevindew) [inline app/models/publish_intent.rb]: yes that'd be better, thanks - (was blindly c+ping methods). @dougdroper would not be happy with the `return nil` there

Comment (thomasleese) [inline app/models/publish_intent.rb]: Indeed. I was making sure we still have the @dougdroper spirit on the team

File: app/controllers/publish_intents_controller.rb [hunk 3]
```diff
@@ -2,19 +2,23 @@ class PublishIntentsController < ApplicationController
...
 
   def update
     result, intent = PublishIntent.create_or_update(encoded_base_path, @request_data)
 
-    if result
-      status = (result == :created ? :created : :ok)
-    else
-      status = :unprocessable_entity
-    end
+    status = if result
```
Comment (kevindew) [inline app/controllers/publish_intents_controller.rb]: it does indeed, good suggestion

**Comment by kevindew** [comment]
@thomasleese comments addressed, thanks

**Pull request merged**